qutip/core/qobj.py

```diff
@@ -1617,7 +1521,28 @@ def eigenstates(
             phase = np.array([np.abs(ket[phase_fix, 0]) / ket[phase_fix, 0]
                               if ket[phase_fix, 0] else 1
                               for ket in ekets])
-        return evals, ekets / norms * phase
+        if output_type == 'ket':
+            return evals, ekets / norms * phase
+        elif output_type == 'oper':
+            ekets = ekets / norms * phase
+
+            eigarr = np.zeros((len(evals), len(evals)), dtype=complex)
+
+            for i in range(len(ekets)):
+                x = ekets[i].full()
+                arr1 = np.array([ele[0] for ele in x])
+                arr1 = arr1.flatten()
```

Comment:
This array manipulation is overly complicated.
You make a 2d (N x 1) array. You loop over the rows in a list comprehension, which you change back to an (1d) array, then flatten it, but it's already flat.

The solution we want does not use numpy array. If it uses `full()`, we will not take that method...